### PR TITLE
Fix/backend completion

### LINKED ISSUE #16596: Fix Back end challenge completed modal
On the back end challenges, make sure a modal pops up and that the user is not advanced to the next challenge until the user clicks the "go to next challenge" button in the modal.

### PR DESCRIPTION
**Pull request by Bouncey**
#### Pre-Submission Checklist

- [x] Your pull request targets the `staging` branch of freeCodeCamp.
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [ ] You have only one commit (if not, [squash](http://forum.freecodecamp.org/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [x] All new and existing tests pass the command `npm test`. Use `git commit --amend` to amend any fixes.

#### Type of Change

- [x] Small bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)

#### Checklist:

- [x] Tested changes locally.
- [x] Addressed currently open issue (replace XXXXX with an issue no in next line)

Closes #16596

### CODE REVIEW

File: common/app/routes/Challenges/redux/completion-epic.js [hunk 1]
```diff
@@ -114,9 +115,14 @@ function submitBackendChallenge(type, state, { solution }) {
...
     );
     */
+    return Observable.empty();
+  }
+  if (type === types.submitChallenge.toString()) {
     const { id } = challengeSelector(state);
     const { username } = userSelector(state);
     const csrfToken = csrfSelector(state);
+    const { solution } = getValues(state.form.BackEndChallenge);
+    console.log(solution)
```
Comment (BerkeleyTrue) [inline common/app/routes/Challenges/redux/completion-epic.js]: console log

File: common/app/routes/Challenges/redux/completion-epic.js [hunk 2]
```diff
@@ -114,9 +115,14 @@ function submitBackendChallenge(type, state, { solution }) {
       })
     );
     */
+    return Observable.empty();
+  }
+  if (type === types.submitChallenge.toString()) {
     const { id } = challengeSelector(state);
     const { username } = userSelector(state);
     const csrfToken = csrfSelector(state);
+    const { solution } = getValues(state.form.BackEndChallenge);
```
Comment (BerkeleyTrue) [inline common/app/routes/Challenges/redux/completion-epic.js]: This should be a selector. Reaching into state is couples this epic with the shape of the state

**Review by BerkeleyTrue — CHANGES_REQUESTED** [review]

**Comment by QuincyLarson** [comment]
@Bouncey once @BerkeleyTrue is happy with this and has dismissed his review I'll QA the functionality and merge :)

**Review by BerkeleyTrue — APPROVED** [review]

**Comment by QuincyLarson** [comment]
LGTM! Thanks, @Bouncey!

**Pull request merged**